src/tests/unit/onbase-dao.test.js

```diff
@@ -1,33 +1,79 @@
-const chai = require('chai');
-const chaiAsPromised = require('chai-as-promised');
-const proxyquire = require('proxyquire');
-const sinon = require('sinon');
-
-const conn = require('../../api/v1/db/oracledb/connection');
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import config from 'config';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+import createError from 'http-errors';
+
+// config must be replaced before connection can be imported
+sinon.replace(config, 'get', () => ({ oracledb: {} }));
+const conn = require('api/v1/db/oracledb/connection');
+const onBaseSerializer = require('api/v1/serializers/onbase-serializer');
 const testData = require('./test-data');
 
-const daosSerializer = require('../../api/v1/serializers/onbase-serializer');
-
 chai.should();
 chai.use(chaiAsPromised);
 
 let onBaseDao;
 
 describe('Test onbase-dao', () => {
-  // the execute function gets called for a variety of reasons in the class being tested
-  // each test will need fine tuned control over what is returned from execute
+  /*
+  * The execute function gets called for a variety of reasons in the class being tested
+  * Each test will need fine tuned control over what is returned from execute
+  */
   const connectionStub = (execStub) => {
     sinon.stub(conn, 'getConnection').resolves({
       execute: execStub,
       close: () => null,
     });
   };
 
+  const testSingleResult = (testFunction) => {
+    const execStub = sinon.stub();
+    execStub.returns(testData.outBindsLast);
+    connectionStub(execStub);
+
+    const result = testFunction();
+    return result.should
+      .eventually.be.fulfilled
+      .and.deep.equals([{}])
+      .and.to.have.length(1);
+  };
+
+  const testMultipleResults = (testFunction) => {
+    const execStub = sinon.stub();
+    // third call is a special case that we want to cause getLine to recurse
+    execStub.onThirdCall().returns(testData.outBindsRecursive);
+    execStub.returns(testData.outBindsLast);
+    connectionStub(execStub);
+
+    const result = testFunction();
+    return result.should
+      .eventually.be.fulfilled
+      .and.deep.equals([{}, {}])
+      .and.to.have.length(2);
+  };
+
+  /*
+    * error message will be the same as the line length
+    * because we are generating an array of numbers
+    */
+  const testLineErrorResult = (testFunction) => {
+    const execStub = sinon.stub();
+    execStub.returns(testData.outBindsError);
+    connectionStub(execStub);
+
+    const result = testFunction();
```

Comment:
Pretty good improvement. One further step, the following codes appear in `testSingleResult`, `testMultipleResults`, and `testLineErrorResult`:

```js
const execStub = sinon.stub();
execStub.returns(testData.outBindsLast); // or returns testData.outBindsError
connectionStub(execStub);

const result = testFunction();
```

The only difference is `testMultipleResults` has an additional block:

```js
// third call is a special case that we want to cause getLine to recurse
execStub.onThirdCall().returns(testData.outBindsRecursive);
```

That being said, I believe we still could still figure a better why to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) the code (maybe a `getTestResult` funciton could help?).